aptos-move/aptos-vm/src/block_executor/mod.rs

```diff
@@ -88,6 +88,10 @@ impl BlockExecutorTransactionOutput for AptosTransactionOutput {
         Self::new(VMOutput::empty_with_status(TransactionStatus::Retry))
     }
 
+    fn discard_output(discard_code: StatusCode) -> Self {
```

Comment:
What's the difference between abort and discard? When do we discard instead of abort?

Reply:
discard means rejected from onchain, i.e. we couldn't even charge the signer gas (for example if sequence number is incorrect, or signature is incorrect, etc)

abort means execution of the transaction failed, but transaction itself gets committed onchain with that status

Reply:
It was not the case before that we would skip from BlockSTM. VM can discard, it creates empty outputs already (e.g., if prologue fails)